plugin/k8s_external/msg_to_dns.go

```diff
@@ -61,10 +61,7 @@ func (e *External) aaaa(services []msg.Service, state request.Request) (records
 	return records
 }
 
-func (e *External) srv(services []msg.Service, state request.Request) (records, extra []dns.RR) {
-	dup := make(map[item]struct{})
-
-	// Looping twice to get the right weight vs priority. This might break because we may drop duplicate SRV records latter on.
+func buildWeights(services []msg.Service) map[int]int {
```

Comment:
do we need this? I.e. can there be multiple msg where we need to calculate srv weights?

Also this change seems to be unrelated to the functionality of this pr, so please split it off and make it a global change in all plugins that do this

Reply:
my PR is to add AXFR, not to change existing SRV function.
So I keep the same computation ... cannot say really how much that is useful or not.

Reply:
but .. sure, I can consider sharing this function in all plugins that have the same computation. Will be in anotehr PR.